### PR TITLE
Support `GENERATED ALWAYS AS expression` in INSERT statement

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Support `GENERATED ALWAYS AS expression` for INSERT statement in engine and Delta Lake connector.
Note that the generated column in Delta Lake isn't virtual (stores the values physically during INSERT). 

```
<generation clause> ::=
  <generation rule>  AS <generation expression> 

<generation rule> ::=
  GENERATED ALWAYS

<generation expression> ::=
  <left paren>  <value expression>  <right paren> 
```

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# General
* Support `GENERATED ALWAYS AS expression` in `INSERT` statement. ({issue}`16565`)
```

### CODE REVIEW

**Comment by dain** [comment]
Is there a connector that will be using this or it this speculative?

File: core/trino-spi/src/main/java/io/trino/spi/connector/GenerationDefinition.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+@Experimental(eta = "2023-06-01")
+public class GenerationDefinition
+{
+    private final GenerationRule rule;
+    private final String expression;
+
+    public GenerationDefinition(GenerationRule rule, String expression)
+    {
+        this.rule = rule;
```
Comment (electrum) [inline core/trino-spi/src/main/java/io/trino/spi/connector/GenerationDefinition.java:26]: Make this class a record and check arguments for null

**Comment by ebyhr** [comment]
> Is there a connector that will be using this or it this speculative?

@dain Delta Lake connector will support it. https://github.com/delta-io/delta/blob/master/PROTOCOL.md#generated-columns

**Comment by ebyhr** [comment]
Pushed another commit supporting Delta Lake connector so that we can understand the SPI usage easily. I can separate a PR if reviewers want me to do that.

**Pull request closed without merging**